### ISSUE TITLE
Feature question for `IncrementalRemesher`: Feature to collapse very triangles with very sharp angle

### ISSUE THREAD

**Issue by Nevsden** [dima634/baby_shark#35]
Does any of the features of the `IncrementalRemesher` collapse whole triangles or collapse two close vertices?
If not, is the implementation of such a feature easy? I could provide this.

**Comment by dima634** [comment]
This should be straightforward to implement. However, are you certain that you need it for IncrementalRemesher and not for EdgeDecimator? The slides discuss incremental decimation, which aligns more closely with what EdgeDecimator implements.

**Comment by Nevsden** [comment]
After doing some research, I came across this algorithm from the RWTH Aachen, which seems to do a good job improving mesh quality after the decimation of edges.

Would this feature be something, you would like to see in your library?


https://www.graphics.rwth-aachen.de/media/papers/slicing1.pdf

**Comment by dima634** [comment]
> @dima634 You are right. This should better be implemented as a feature to the `EdgeDecimator` as a plug-in criterion. E.g.
> 
> ```
>     // Quality taken from 
>     let quality = radius / edge_length;
>     let mut decimator = EdgeDecimator::new()
>         .decimation_criteria(decimation_criteria)
>         .keep_boundary(true)
>         .enforce_triangle_shape(quality);
>     decimator.decimate(&mut mesh);
> ```

I think the easiest way to implement it is to create a new [`EdgeDecimationCriteria`](https://github.com/dima634/baby_shark/blob/cdfa04d784a71717cc9758d5509e6408810e9544/src/decimation/edge_decimation.rs#L387) that would check triangle quality after collapse. And if it drops below a certain value ban such edge collapse.

**Comment by dima634** [comment]
> After doing some research, I came across this algorithm from the RWTH Aachen, which seems to do a good job improving mesh quality after the decimation of edges.
> 
> Would this feature be something, you would like to see in your library?
> 
> https://www.graphics.rwth-aachen.de/media/papers/slicing1.pdf

I find it quite similar to what `IncrementalRemesher` currently offers (however sharp feature may be smoothed a bit, not sure).

**Comment by Nevsden** [comment]
You are right, it's very similar. But in one point I would contradict: the algorithm described in the paper still tries to keep the number of edges minimal. 

Out of curiosity: would this minimum edges feature be something that I could already achieve with the Remesher pipeline?
